### ISSUE TITLE
Finetune or training models

### ISSUE THREAD

**Issue by ovesreinier** [indranaut/3DCNNs#1]
I would like to thank first for your work in rewriting the I3D for keras framework
Do you have a code to finetune your model ?
Do you have any pretrained model over kinetics dataset with similar results as the authors report ?
Regards, Reinier

**Comment by indranaut** [comment]
Yes. I am currently testing the code by training I3D on Kinetics-700 dataset ( after inflating the model with Imagenet weights). 
The training is currently running and as soon as it finishes, I will push following items to the repository
a) Training code (can also be used for finetuning).
b) Link to pre-trained model.

**Comment by ovesreinier** [comment]
Sorry.
But, can you share the training code ?
The problem is that the model you shared, prints 0 parameters when
summary...

Regards

**Comment by indranaut** [comment]
@ovesreinier  There is nothing wrong with the model code.
It prints 0 parameters, because you have not called the model yet. 
model.summary() does not return any parameters unless all the layers within it have been built. 

To achieve this, you need to call the build() method as follows. Let me know

```
from nets.i3d import Inception3D


net = Inception3D()
net.build(input_shape=(None,64,224,224,3))
print(net.summary())
```

**Comment by indranaut** [comment]
@ovesreinier With regards to the training code, rest assured that I shall share it here. This is the purpose of this repository.
I will however upload it only after I have run all the tests with the code. 
I do not want the users of this repository to go through issues due to untested code. 
Over the next 2 weeks you can expect the fully functioning training code and a model of I3D trained by me.
Thank you for your understanding and support.

**Comment by ovesreinier** [comment]
Thanks a lot.
I appreciate your contribution.
I tested your code and 0 parameters still printing
Don't worry I understand
Thanks one more time
Regards. Reinier

**Comment by indranaut** [comment]
@ovesreinier  
I share the code below and its execution as well. 

Can you share your TensorFlow version exactly ? Is it TF2.0 ?

```
(tensorflow2.0master) -bash-4.2$ cat main.py 
from nets.i3d import Inception3D


net = Inception3D()
net.build(input_shape=(None,64,224,224,3))
print(net.summary())

(tensorflow2.0master) -bash-4.2$ python main.py 
2019-08-20 18:55:06.154996: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2019-08-20 18:55:08.365552: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1
2019-08-20 18:55:08.391008: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: 
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.7715
pciBusID: 0000:02:00.0
2019-08-20 18:55:08.391045: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2019-08-20 18:55:08.393122: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
2019-08-20 18:55:08.394693: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0
2019-08-20 18:55:08.395183: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0
2019-08-20 18:55:08.397138: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0
2019-08-20 18:55:08.398626: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0
2019-08-20 18:55:08.402196: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
2019-08-20 18:55:08.403840: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
2019-08-20 18:55:08.554967: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5558e20c6140 executing computations on platform CUDA. Devices:
2019-08-20 18:55:08.555019: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): GeForce GTX 1080, Compute Capability 6.1
2019-08-20 18:55:08.557484: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2197470000 Hz
2019-08-20 18:55:08.558759: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5558e2139560 executing computations on platform Host. Devices:
2019-08-20 18:55:08.558778: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2019-08-20 18:55:08.560149: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: 
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.7715
pciBusID: 0000:02:00.0
2019-08-20 18:55:08.560177: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2019-08-20 18:55:08.560198: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
2019-08-20 18:55:08.560210: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0
2019-08-20 18:55:08.560220: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0
2019-08-20 18:55:08.560230: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0
2019-08-20 18:55:08.560241: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0
2019-08-20 18:55:08.560252: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
2019-08-20 18:55:08.561669: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
2019-08-20 18:55:08.561692: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2019-08-20 18:55:08.921363: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-08-20 18:55:08.921420: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 
2019-08-20 18:55:08.921430: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N 
2019-08-20 18:55:08.923665: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7512 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080, pci bus id: 0000:02:00.0, compute capability: 6.1)
Model: "inception_i3d"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
MainBranch (Unit3D)          multiple                  66112     
_________________________________________________________________
MaxPool3d_2a_3x3 (MaxPooling multiple                  0         
_________________________________________________________________
MainBranch (Unit3D)          multiple                  4352      
_________________________________________________________________
MainBranch (Unit3D)          multiple                  332544    
_________________________________________________________________
MaxPool3d_3a_3x3 (MaxPooling multiple                  0         
_________________________________________________________________
Mixed_3b/Branch_0 (Unit3D)   multiple                  12544     
_________________________________________________________________
Mixed_3b/Branch_1 (Unit3D)   multiple                  18816     
_________________________________________________________________
Mixed_3b/Branch_1 (Unit3D)   multiple                  332288    
_________________________________________________________________
Mixed_3b/Branch_2 (Unit3D)   multiple                  3136      
_________________________________________________________________
Mixed_3b/Branch_2 (Unit3D)   multiple                  13952     
_________________________________________________________________
Mixed_3b/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_3b/Branch_3 (Unit3D)   multiple                  6272      
_________________________________________________________________
Mixed_3c/Branch_0 (Unit3D)   multiple                  33280     
_________________________________________________________________
Mixed_3c/Branch_1 (Unit3D)   multiple                  33280     
_________________________________________________________________
Mixed_3c/Branch_1 (Unit3D)   multiple                  664320    
_________________________________________________________________
Mixed_3c/Branch_2 (Unit3D)   multiple                  8320      
_________________________________________________________________
Mixed_3c/Branch_2 (Unit3D)   multiple                  83328     
_________________________________________________________________
Mixed_3c/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_3c/Branch_3 (Unit3D)   multiple                  16640     
_________________________________________________________________
Mixed_4aMaxPool3d_0a_3x3 (Ma multiple                  0         
_________________________________________________________________
Mixed_4b/Branch_0 (Unit3D)   multiple                  92928     
_________________________________________________________________
Mixed_4b/Branch_1 (Unit3D)   multiple                  46464     
_________________________________________________________________
Mixed_4b/Branch_1 (Unit3D)   multiple                  539968    
_________________________________________________________________
Mixed_4b/Branch_2 (Unit3D)   multiple                  7744      
_________________________________________________________________
Mixed_4b/Branch_2 (Unit3D)   multiple                  20928     
_________________________________________________________________
Mixed_4b/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_4b/Branch_3 (Unit3D)   multiple                  30976     
_________________________________________________________________
Mixed_4c/Branch_0 (Unit3D)   multiple                  82560     
_________________________________________________________________
Mixed_4c/Branch_1 (Unit3D)   multiple                  57792     
_________________________________________________________________
Mixed_4c/Branch_1 (Unit3D)   multiple                  678272    
_________________________________________________________________
Mixed_4c/Branch_2 (Unit3D)   multiple                  12384     
_________________________________________________________________
Mixed_4c/Branch_2 (Unit3D)   multiple                  41728     
_________________________________________________________________
Mixed_4c/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_4c/Branch_3 (Unit3D)   multiple                  33024     
_________________________________________________________________
Mixed_4d/Branch_0 (Unit3D)   multiple                  66048     
_________________________________________________________________
Mixed_4d/Branch_1 (Unit3D)   multiple                  66048     
_________________________________________________________________
Mixed_4d/Branch_1 (Unit3D)   multiple                  885760    
_________________________________________________________________
Mixed_4d/Branch_2 (Unit3D)   multiple                  12384     
_________________________________________________________________
Mixed_4d/Branch_2 (Unit3D)   multiple                  41728     
_________________________________________________________________
Mixed_4d/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_4d/Branch_3 (Unit3D)   multiple                  33024     
_________________________________________________________________
Mixed_4e/Branch_0 (Unit3D)   multiple                  57792     
_________________________________________________________________
Mixed_4e/Branch_1 (Unit3D)   multiple                  74304     
_________________________________________________________________
Mixed_4e/Branch_1 (Unit3D)   multiple                  1120896   
_________________________________________________________________
Mixed_4e/Branch_2 (Unit3D)   multiple                  16512     
_________________________________________________________________
Mixed_4e/Branch_2 (Unit3D)   multiple                  55552     
_________________________________________________________________
Mixed_4e/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_4e/Branch_3 (Unit3D)   multiple                  33024     
_________________________________________________________________
Mixed_4f/Branch_0 (Unit3D)   multiple                  136192    
_________________________________________________________________
Mixed_4f/Branch_1 (Unit3D)   multiple                  85120     
_________________________________________________________________
Mixed_4f/Branch_1 (Unit3D)   multiple                  1383680   
_________________________________________________________________
Mixed_4f/Branch_2 (Unit3D)   multiple                  17024     
_________________________________________________________________
Mixed_4f/Branch_2 (Unit3D)   multiple                  111104    
_________________________________________________________________
Mixed_4f/Branch_2/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_4f/Branch_3 (Unit3D)   multiple                  68096     
_________________________________________________________________
Mixed_5a/MaxPool3d_0a_3x3 (M multiple                  0         
_________________________________________________________________
Mixed_5b/Branch_0 (Unit3D)   multiple                  214016    
_________________________________________________________________
Mixed_5b/Branch_1 (Unit3D)   multiple                  133760    
_________________________________________________________________
Mixed_5b/Branch_1 (Unit3D)   multiple                  1383680   
_________________________________________________________________
Mixed_5b/Branch_2 (Unit3D)   multiple                  26752     
_________________________________________________________________
Mixed_5b/Branch_2 (Unit3D)   multiple                  111104    
_________________________________________________________________
Mixed_5b/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_5b/Branch_3 (Unit3D)   multiple                  107008    
_________________________________________________________________
Mixed_5c/Branch_0 (Unit3D)   multiple                  321024    
_________________________________________________________________
Mixed_5c/Branch_1 (Unit3D)   multiple                  160512    
_________________________________________________________________
Mixed_5c/Branch_1 (Unit3D)   multiple                  1992192   
_________________________________________________________________
Mixed_5c/Branch_2 (Unit3D)   multiple                  40128     
_________________________________________________________________
Mixed_5c/Branch_2 (Unit3D)   multiple                  166400    
_________________________________________________________________
Mixed_5c/Branch_3/MaxPool3d_ multiple                  0         
_________________________________________________________________
Mixed_5c/Branch_3 (Unit3D)   multiple                  107008    
_________________________________________________________________
avgpool (AveragePooling3D)   multiple                  0         
_________________________________________________________________
dropout (Dropout)            multiple                  0         
_________________________________________________________________
Logits (Unit3D)              multiple                  410000    
_________________________________________________________________
lambda (Lambda)              multiple                  0         
_________________________________________________________________
lambda_1 (Lambda)            multiple                  0         
=================================================================
Total params: 12,711,824
Trainable params: 12,697,264
Non-trainable params: 14,560
_________________________________________________________________
None
(tensorflow2.0master) -bash-4.2$ 

```

**Comment by indranaut** [comment]
And here is my Python environment and my TensorFlow version

```
(tensorflow2.0master) -bash-4.2$ ipython
Python 3.6.7 | packaged by conda-forge | (default, Jul  2 2019, 02:18:42) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.6.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import tensorflow as tf                                                                                                                                                                                    
2019-08-20 18:57:43.301196: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
tf.__v
In [2]: tf.__version__                                                                                                                                                                                             
Out[2]: '2.0.0-beta1'

In [3]:           
```

**Comment by ovesreinier** [comment]
Sorry, I am using the 1.12.3 tf version

Regards, Reinier

**Comment by indranaut** [comment]
This repository is specifically aimed at TF2.0. 

If explicit support for versions before 2.0 is also needed, open a separate issue and let me know. based on people's demands it might be a worthwhile feature to have.
